kv/README.md

```diff
@@ -0,0 +1,84 @@
+# Proposed design for JetStream based KV services
+
+This is a proposal to design KV services ontop of JetStream, this requires a KV client to be written, the basic
+features are:
+
+ * Multiple named buckets full of keys with n historical values kept per value
+ * Put and Get `string(k)=string(v)` values
```

Comment:
Iteration is hard given how JS works underneath, I want iteration but parked it a bit till later.

To iterate means loading the entire bucket into memory, filling out a map - or keeping a list - and then iterating that list.  This can be quite expensive on big buckets. I think we might need a API for asking JS for a list of subjects in a stream.  So parked iteration for now.  It's not a very common thing in kv stores in my experience - though some do have it.

Reply:
we could use a secondary sorted index on disk

Reply:
Which disk? In JS? sure - Derek is looking at adding some capabilities, for now with what we have, no good.